src/basic/pager.h

```diff
@@ -15,3 +15,9 @@ void pager_close(void);
 bool pager_have(void) _pure_;
 
 int show_man_page(const char *page, bool null_stdio);
+
+/* A dummy struct to make _cleanup_ type-safe */
+typedef struct Pager {} Pager;
+static inline void pager_closep(Pager *pager) {
+        pager_close();
+}
```

Comment:
hmm, i wonder if this is the best way to model this. the pager really should be closed last, since otherwise stdout/stderr will have to go back to the old stdou/stderr, which would be really weird behaviour. Or in other words I think we should really make sure that the pager is not just started within some specific scope (which is what the cleanup stuff in your patch does after all, it scopes the pager till the next `}`) but instead is always started until the very end of the program. Hence I wonder if it wouldn't be more appropriate to instead extend the static destructor stuff to make this work, as for that we do know when things are destructed: right before exiting. Or alternatively we could adjust the DEFINE_MAIN_FUNC macros to call pager_close() implicitly.